tests/test_starfileio.py

```diff
@@ -1,25 +1,56 @@
+import importlib_resources
+import numpy as np
 import os.path
+
+from itertools import zip_longest
+from os.path import splitext
 from unittest import TestCase
 
 import importlib_resources
+import numpy as np
 from pandas import DataFrame
+from scipy import misc
 
 import tests.saved_test_data
-from aspire.io.starfile import StarFile, StarFileBlock
+from aspire.image import Image
+from aspire.io.starfile import StarFile, StarFileBlock, save_star
+from aspire.source import ArrayImageSource
+from aspire.source.mrcstack import MrcStack
 
 DATA_DIR = os.path.join(os.path.dirname(__file__), 'saved_test_data')
 
 
+# From itertools standard recipes
+def grouper(iterable, n, fillvalue=None):
```

Comment:
This can be a little opaque to readers. Can you add a quick description (just take the one from the recipes)?

Reply:
Yep, you're right. 45840a9ace5c42a83941eecd68c8ec09aff5b476